File: server/order-service/src/main/java/com/sudo248/orderservice/controller/order/dto/OrderSupplierUserInfoDto.java
package com.sudo248.orderservice.controller.order.dto;

import com.sudo248.orderservice.repository.entity.order.OrderStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class OrderSupplierUserInfoDto {
    private String orderSupplierId;
    private String supplierId;
    private String supplierName;
    private String supplierAvatar;
    private String supplierBrand;
    private String supplierContactUrl;
    private OrderStatus status;
    private LocalDateTime expectedReceiveDateTime;
    private double totalPrice;
    private List<OrderCartProductDto> orderCartProducts;
}

File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/data/dto/order/OrderSupplierUserInfoDto.kt
package com.sudo248.sudoo.data.dto.order

import com.sudo248.base_android.base.ItemDiff
import com.sudo248.sudoo.domain.entity.order.OrderStatus
import com.sudo248.sudoo.domain.entity.order.OrderSupplier
import java.time.LocalDateTime


data class OrderSupplierUserInfoDto(
    private val orderSupplierId: String,
    private val supplierId: String,
    private val supplierName: String,
    private val supplierAvatar: String,
    private val supplierBrand: String,
    private val supplierContactUrl: String,
    private val status: OrderStatus,
    private val expectedReceiveDateTime: LocalDateTime,
    private val totalPrice: Double = 0.0,
    private val orderCartProducts: List<OrderCartProductDto>,
): ItemDiff {
    override fun isContentTheSame(other: ItemDiff): Boolean {
        return other is OrderSupplier && other == this
    }

    override fun isItemTheSame(other: ItemDiff): Boolean {
        return other is OrderSupplier && other.orderSupplierId == orderSupplierId
    }

}